serverlessworkflow/modules/ROOT/pages/tooling/serverless-workflow-editor/swf-editor-vscode-extension.adoc

```diff
@@ -0,0 +1,182 @@
+
+== Working with VScode Extension
+
+Once installation is complete, you can create a Serverless Workflow file or open an existing file in VScode. VScode will automatically use the extension when you open a Serverless Workflow file.
+
+The extension provides a set of commands to perform specific actions.
+
+=== List of commands
+* *Serverless Workflow: Generate SVG without any notification*: Generates an SVG file of the diagram in the workspace next to the Serverless Workflow file.
+* *Serverless Workflow: Open as Diagram (to the side)*: Opens the diagram to the right side of the editor
+* *Serverless Workflow: Setup automatically open Diagram Editor alongside Text Editor*: Allows the user to configure if the diagram should be opened by default alongside the editor. Provides two options, *Open automatically* and *Do not open*.
+* *Serverless Workflow: Configure Service Registries*: Opens the configuration for the Service Registries.
+* *Serverless Workflow: Refresh Service Registries*: Refresh the artifact list of the Service Registries.
+* *Serverless Workflow: Log in Service Registries*: Triggers a log in action of the Service Registries.
+
+To trigger a command, go to *view* → *Command Palette*... or press *Ctrl+Shift+P* to open the command palette. Type in *Serverless Workflow* in the search bar to see the list of commands.
+
+.Command palette of the VScode extension
+image::tooling/serverless-workflow-editor/swf-editor-vscode-command-palette.png[]
+
+=== Kogito Serverless Workflow Editor Setttings
+The extension's settings can be configured by navigating to *File* → *Preferences* → *Settings*.
+
+.The following settings can be configured in the settings page:
+*  *Automatically Open Diagram Editor Alongside Text Editor*: Provides a dropdown with *Ask next time*(default), *Do not open* and *Open automatically* to configure the opening of the diagram.
+* *Run On Save*: Executes a command to generate an SVG of the workflow when a _save_ action is performed.
+* *Service Registries*: Provides a link to access the _settings.json_, to configure the Serivce Registries.
+* *Should Reference Service Registry Functions With Urls*: Provides a check box to toggle between downloading the openAPI file or use the url of the function in Service Registry.
+* *Specs Storage Path*: Provides a text box to edit the path of openAPI spec file(defaults to 'specs' folder).
+* *Svg Filename Template*: Provides a text box to edit the filename template to be used when generating SVG files.
+* *Svg File Path*: Provides a text box to edit the path to generate SVG files.
+
+.Settings page of the extension
+image::tooling/serverless-workflow-editor/swf-editor-vscode-settings.png[]
+
+=== Context based auto-completion
+The extension can use openAPI spec files stored locally in the 'specs folder'(<<Kogito Serverless Workflow Editor Setttings,check the settings section>>) or in remote Service Registries to provide advanced auto-complete features for *Function Definitions*.
```

Comment:
Can you please add a link to the OpenAPI Spec? There's a variable introduced by this PR: #18